### PR TITLE
refactor: add missing typings

### PR DESCRIPTION
**Pull request by ChristianMurphy**
Typescript's `any` feature disables type checking for a variable, this change goes through areas that have been disabled and adds appropriate types.

### CODE REVIEW

**Comment by edalex-ian** [comment]
Looks good, but unfortunately we don't have test coverage on some of these areas I fear. :thinking: 

At least one bit of this code is yours @SammyIsConfused , so please review.

The the question is how to we test it. Seems the areas of interest would be:

* Course editor
* General display of errors
* And places validating integers
* Plus some search results - which is curious, as I thought the current New Search UI was all PS (but maybe this is something else)

Did you have a chance to test any of this @ChristianMurphy ?

**Comment by ChristianMurphy** [comment]
I tested this by running `npm run build`.
The changes updates the typescript typings, which exist only during compile time and do not affect the run time.

The one change that could affect the run time is `validation.ts`, which I checked with a truth table.
But have not done extensive testing of course service.

**Comment by ChristianMurphy** [comment]
ping @edalex-ian :bell:

File: Source/Plugins/Core/com.equella.core/js/tsrc/util/validation.ts [hunk 1]
```diff
@@ -1,16 +1,12 @@
 export function isInteger(
-  val: any,
+  val: number | undefined,
```
Comment (abidingotter) [inline Source/Plugins/Core/com.equella.core/js/tsrc/util/validation.ts]: How about `val?: number`

File: Source/Plugins/Core/com.equella.core/js/tsrc/components/index.ts [hunk 2]
```diff
@@ -7,10 +7,7 @@ export { default as Loader } from "./Loader";
...
 export { default as SearchResult } from "./SearchResult";
 
-export function handleUnexpectedApiError<P extends TemplateUpdateProps>(
-  t: React.Component<P>
-): (err: any) => void {
-  return function(err) {
-    t.props.updateTemplate(templateError(generateFromError(err)));
-  };
-}
+export const handleUnexpectedApiError = <P extends TemplateUpdateProps>({
```
Comment (abidingotter) [inline Source/Plugins/Core/com.equella.core/js/tsrc/components/index.ts]: I would argue that `export const` is less readable than `export function`

Comment (ChristianMurphy) [reply Source/Plugins/Core/com.equella.core/js/tsrc/components/index.ts]: Fair, my tendency is to use arrow functions in any situation where I don't need to [`bind`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind) the `function`.
I'd be fine walking that back to a `function` if you feel strongly that `function` is more readable.

Comment (abidingotter) [reply Source/Plugins/Core/com.equella.core/js/tsrc/components/index.ts]: I'm not that concerned about it :)  Just thought that `export function` tells you right up what is being exported, without you having to decide what it is.  I agree about use of arrow functions elsewhere.

File: Source/Plugins/Core/com.equella.core/js/tsrc/util/validation.ts (hunk 1 above)
Comment (ChristianMurphy) [reply Source/Plugins/Core/com.equella.core/js/tsrc/util/validation.ts]: Updated

File: Source/Plugins/Core/com.equella.core/js/tsrc/components/index.ts (hunk 2 above)
Comment (ChristianMurphy) [reply Source/Plugins/Core/com.equella.core/js/tsrc/components/index.ts]: switched to `function`

**Pull request merged**